feat(grid): implement functionality for deleting a row

hdimitrieski opened the pull request.

Commit:
feat(grid): implement functionality for deleting a row

velinovskav commented on projects/data-grid/src/lib/store/data-grid.ts:
```diff
@@ -250,6 +244,26 @@ const toggleDetailGrid = (state: GridState, {child, active}: ToggleDetailsGridPa
+      rowDataIndexes: R.remove(dataItemIndex, 1),
+      data: R.remove(dataItemIndex, 1)
+    }, state) : state;
+  }
```
In order to avoid code duplication, isn't better first to resolve index of data item and after that to perform the state update.

Commit:
refactor: extract function

Extract the code for resolving item index in a separate function.
Add test cases to cover the cases when the row that does not
exists is being deleted.
